### PR TITLE
feat: added multiplexing for zapUrls

### PR DESCRIPTION
**Pull request by aanshi07**
## What are the changes introduced in this PR?

Mapping RS event to multiple Zap URLs

## What is the related Linear task?

Resolves INT-2346

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

@coderabbitai review

<hr>

### Developer checklist

- [ ] My code follows the style guidelines of this project

- [ ] **No breaking changes are being introduced.**

- [ ] All related docs linked with the PR?

- [ ] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [ ] Is the PR limited to 10 file changes?

- [ ] Is the PR limited to one linear task?

- [ ] Are relevant unit and component test-cases added in **new readability format**?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3759/test-report.html

File: src/cdk/v2/destinations/zapier/procWorkflow.yaml [hunk 1]
```diff
@@ -37,10 +36,5 @@ steps:
...
       name: buildResponseForProcessTransformation
       template: |
-        const response = $.defaultRequestConfig();
-        response.body.JSON = .message;
-        response.endpoint = $.context.endpoint;
-        response.headers = {
-          "content-type": "application/json"
-        };
-        response
+        const responseList = $.buildResponseList(.message, $.context.endpoint, { "content-type": "application/json" })
```
Comment (manish339k) [inline src/cdk/v2/destinations/zapier/procWorkflow.yaml:39]: instead of passing header, you can directly use it in buildResponseList method.

**Review by manish339k — APPROVED** [review]

File: test/integrations/destinations/zapier/processor/data.ts [hunk 2]
```diff
@@ -703,4 +759,93 @@ export const data = [
       },
     },
   },
+  {
+    name: 'zapier',
+    description: 'Test 5',
+    feature: 'processor',
```
Comment (anantjain45823) [inline test/integrations/destinations/zapier/processor/data.ts:765]: test 5 should be id and description should be something around what scenario test case is describing or summary of test case.
include scenario as well here

File: test/integrations/destinations/zapier/processor/data.ts [hunk 3]
```diff
@@ -703,4 +759,93 @@ export const data = [
...
+            statusCode: 200,
+            metadata: {
+              destinationId: 'destId',
+              workspaceId: 'wspId',
+            },
+          },
+        ],
+      },
+    },
+  },
```
Comment (anantjain45823) [inline test/integrations/destinations/zapier/processor/data.ts:850]: I don't see a test case for page call multiplexing can we add that please ?

**Review by shrouti1507 — APPROVED** [review]

**Review by manish339k — APPROVED** [review]

**Comment by sanpj2292** [comment]
@aanshi07 can we close this PR ? As #3825 is merged already

**Pull request closed without merging**